tamr_toolbox/project/_common/schema.py

```diff
@@ -43,6 +48,113 @@ def _get_mapping_spec_for_ud(
     return spec
 
 
+def create_unified_attribute(project: Project, *, unified_attribute_name: str) -> Attribute:
+    """
+    Adds a unified attribute to a project
+
+    Args:
+        project: Tamr project in which to create a unified attribute
+        unified_attribute_name: Name of the unified attribute
+
+    Returns:
+        The created Attribute
+
+    Raises:
+        AttributeError if the unified attribute already exists
+    """
+    if unified_attribute_name in [x.name for x in project.attributes]:
+        error_message = (
+            f"A unified attribute with name {unified_attribute_name} already exists in "
+            f"{project.name}. Please try again using a new name for the attribute."
+        )
+        LOGGER.error(error_message)
+        raise AttributeError(error_message)
+
+    # Create a new attribute in the project unified dataset
+    attr_spec = {
+        "name": unified_attribute_name,
+        "type": {"baseType": "ARRAY", "innerType": {"baseType": "STRING"}},
+    }
+    return project.attributes.create(attr_spec)
```

Comment:
Do we really want a `create_unified_attribute` function? What's wrong with `project.attributes.create()`?  Is the benefit
- avoid creation of attributes on the unified dataset, rather than the project
- use the default array<str> type (this could just be enumerated under `types`, along with geo_type
- just being functional vs. OO

Reply:
I'm removing this from the PR.  This work is worth revisiting, but doesn't need to be a part of this ticket.